File: Programmers/타겟넘버.cpp
#include <string>
#include <vector>

using namespace std;

int answer = 0;

void calc(vector<int> numbers, int index, int sum, int target) {
    if(index == numbers.size()) {
        if(sum == target) answer++;
        return;
    }
    calc(numbers, index+1, sum+numbers[index], target);
    calc(numbers, index+1, sum-numbers[index], target);
}

int solution(vector<int> numbers, int target) {
    calc(numbers, 0, 0, target);
    return answer;
}

File: Programmers/타겟넘버.java
class 타겟넘버 {
    public static int answer = 0;
    public void getTargetCnt(int[] numbers, int index, int sum, int target) {
        if(index == numbers.length) {
            if(sum == target) answer++;
            return;
        }
        getTargetCnt(numbers, index+1, sum+numbers[index], target);
        getTargetCnt(numbers, index+1, sum-numbers[index], target);
    }
    public int solution(int[] numbers, int target) {
        getTargetCnt(numbers, 0, 0, target);
        return answer;
    }
}